ezomero/json_api.py

```diff
@@ -0,0 +1,248 @@
+
+    session : ``requests`` Session object or None
+        The effective ``requests`` session that will be used for further
+        requests to the JSON API.
+
+    base_url : str or None
+        Base URL for further requests, retrieved via JSON API request
+
+    Notes
+    -----
+    The procedure for choosing parameters for initializing a JSON API session
+    is as follows:
+
+    1) Any parameters given to `create_json_connection` will be used to
+       initialize a JSON session
+
+    2) If a parameter is not given to this function, populate from variables
+       in ``os.environ``:
+        * OMERO_USER
+        * OMERO_PASS
+        * OMERO_WEB_HOST
+
+    3) If environment variables are not set, try to load from a config file.
+       This file should be called '.ezomero'. By default, this function will
+       look in the home directory, but ``config_path`` can be used to specify
+       a directory in which to look for '.ezomero'.
+
+       The function ``ezomero.store_connection_params`` can be used to create
+       the '.ezomero' file.
+
+       Note that passwords can not be loaded from the '.ezomero' file. This is
+       to discourage storing credentials in a file as cleartext.
+
+    4) If any remaining parameters have not been set by the above steps, the
+       user is prompted to enter a value for each unset parameter.
+    """
+    # load from .ezomero config file if it exists
+    if config_path is None:
+        config_fp = Path.home() / '.ezomero'
+    elif type(config_path) is str:
+        config_fp = Path(config_path) / '.ezomero'
+    else:
+        raise TypeError('config_path must be a string')
+
+    config_dict = {}
+    if config_fp.exists():
+        config = configparser.ConfigParser()
+        with config_fp.open() as fp:
+            config.read_file(fp)
+        try:
+            config_dict = config["JSON"]
+        except KeyError:
+            raise KeyError('.ezomero does not contain JSON information.')
+
+    # set user
+    if user is None:
+        user = config_dict.get("OMERO_USER", user)
+        user = os.environ.get("OMERO_USER", user)
+    if user is None:
+        user = input('Enter username: ')
+
+    # set password
+    if password is None:
+        password = os.environ.get("OMERO_PASS", password)
+    if password is None:
+        password = getpass('Enter password: ')
+
+    # set web host
+    if web_host is None:
+        web_host = config_dict.get("OMERO_WEB_HOST", web_host)
+        web_host = os.environ.get("OMERO_WEB_HOST", web_host)
+    if web_host is None:
+        web_host = input('Enter host: ')
+
+    session = requests.Session()
+    # Start by getting supported versions from the base url...
+    api_url = '%s/api/' % web_host
+    r = session.get(api_url, verify=verify)
+    # we get a list of versions
+    versions = r.json()['data']
+    # use most recent version...
+    version = versions[-1]
```

Comment:
I would mention that the api will have multiple versions and that this function will always use the latest. This _possibly_ should be something that can be specified as a parameter

Reply:
mention _in the docstring_

Reply:
I think docstring mention should be enough - if you're in deep enough to be worried about which version you're using, this is probably not very ezomero-y